app/src/main/java/com/practicum/playlistmaker/searchRecyclerPack/RecyclerSearchAdapter.kt

```diff
@@ -10,12 +11,16 @@ import androidx.recyclerview.widget.RecyclerView
 import com.bumptech.glide.Glide
 import com.bumptech.glide.load.resource.bitmap.RoundedCorners
 import com.practicum.playlistmaker.R
+import com.practicum.playlistmaker.SearchHistoryManager
 import com.practicum.playlistmaker.dpToPx
 import java.text.SimpleDateFormat
 import java.util.Locale
 
-class RecyclerSearchAdapter(private val context: Context, private val tracks: List<Track>):
-    RecyclerView.Adapter<RecyclerSearchAdapter.TrackHolder>() {
+class RecyclerSearchAdapter(
+    private val context: Context,
+    private var tracks: MutableList<Track>,
+    private val searchHistoryManager: SearchHistoryManager?
+) : RecyclerView.Adapter<RecyclerSearchAdapter.TrackHolder>() {
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TrackHolder {
         val itemView = LayoutInflater.from(context).inflate(R.layout.search_list_item, parent, false)
```

Comment:
Чтобы контекст не передавать, здесь можешь заменить следующим образом: 
```suggestion
        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.search_list_item, parent, false)
```